ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
Oh, it looks like you're compiling as C90. If you compile as C99 there shouldn't be any warnings. Is that an option?